osu.Game/Overlays/Mods/AddPresetButton.cs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Database;
+using osu.Game.Extensions;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+using osuTK;
+using osu.Game.Localisation;
+
+namespace osu.Game.Overlays.Mods
+{
+    public class AddPresetButton : ShearedToggleButton, IHasPopover
+    {
+        [Resolved]
+        private OsuColour colours { get; set; } = null!;
+
+        [Resolved]
+        private Bindable<IReadOnlyList<Mod>> selectedMods { get; set; } = null!;
+
+        public AddPresetButton()
+            : base(1)
+        {
+            RelativeSizeAxes = Axes.X;
+            Height = ModSelectPanel.HEIGHT;
+
+            // shear will be applied at a higher level in `ModPresetColumn`.
+            Content.Shear = Vector2.Zero;
+            Padding = new MarginPadding();
+
+            Text = "+";
+            TextSize = 30;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            selectedMods.BindValueChanged(val => Enabled.Value = val.NewValue.Any(), true);
+            Enabled.BindValueChanged(val =>
+            {
+                if (!val.NewValue)
+                    Active.Value = false;
+            });
```

Comment:
Super minor, but I generally prefer naming these with context:

```suggestion
            selectedMods.BindValueChanged(mods => Enabled.Value = mods.NewValue.Any(), true);
            Enabled.BindValueChanged(enabled =>
            {
                if (!enabled.NewValue)
                    Active.Value = false;
            });
```